addons/point_of_sale/static/src/app/printer/base_printer.js

```diff
@@ -66,23 +66,24 @@ export class BasePrinter {
      * Renders the html as an image to print it
      * @param {String} receipt: The receipt to be printed, in HTML
      */
-    htmlToImg(htmlEl) {
+    async htmlToImg(htmlEl) {
         const receipt = htmlEl.cloneNode(true);
         const receiptContainer = document.querySelector(".pos-receipt-print");
         receiptContainer.appendChild(receipt);
-        this.receipt = $(receiptContainer.querySelector(".pos-receipt"));
+        this.receipt = receiptContainer.querySelector(".pos-receipt");
         // Odoo RTL support automatically flip left into right but html2canvas
         // won't work as expected if the receipt is aligned to the right of the
         // screen so we need to flip it back.
-        this.receipt.parent().css({ left: 0, right: "auto" });
-        return html2canvas(this.receipt[0], {
-            height: Math.ceil(this.receipt.outerHeight() + this.receipt.offset().top),
-            width: Math.ceil(this.receipt.outerWidth() + 2 * this.receipt.offset().left),
+        const parent = this.receipt.parentElement;
+        parent.style.left = 0;
+        parent.style.right = "auto";
+        const canvas = await html2canvas(this.receipt, {
+            height: Math.ceil(this.receipt.scrollHeight + this.receipt.offsetTop),
+            width: Math.ceil(this.receipt.scrollWidth + 2 * this.receipt.offsetLeft),
```

Comment:
@sdegueldre

During my debugging, I noticed that `this.receipt.offset().top` and `this.receipt.offset().left` are always zero. I start to think that they are all true for our use cases because the ".pos-receipt" wrapper is not positioned at all against its parent (".pos-receipt-print") so its offsets will be zero.

Do you think I can just remove the terms `this.receipt.offsetTop` and `2 * this.receipt.offsetLeft` for the `height` and `width`?

Reply:
And oh, I should use clientHeight and clientWidth. I don't think there is scrolling in the htmlElement that we pass to the html2canvas function.